### PR TITLE
Newsletter notifications

### PR DESCRIPTION
**Pull request by dmolineus**
After #156 was rejected in the end, this is the follow up PR providing notifications for the newsletter system of Contao.

This PR starts from the current newsletter module implementation of Contao 4.6 and provides backwards compatibility layer for Contao < 4.1. I refactored the module in small steps/commits so that you can review the changes.

The notifications are implemented this way:

- Different modules for subscribe, activate, unsubscribe
- Custom modules only sending notifications
- Using different notification types, as subscribe requires different tokens (token and link)

/cc @frontendschlampe

### CODE REVIEW

File: modules/ModuleNewsletterUnsubscribeNotificationCenter.php [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+        {
+            $_SESSION['UNSUBSCRIBE_CONFIRM'] = $GLOBALS['TL_LANG']['MSC']['nl_removed'];
+        }
+
+        $this->reload();
+    }
+
+    protected function compileConfirmationMessage()
+    {
+        if (version_compare(VERSION, '4.1', '>='))
```
Comment (Toflar) [inline modules/ModuleNewsletterUnsubscribeNotificationCenter.php]: I think we could raise the minimum dependencies to Contao 3.5 or 4.4, all the others are unmaintained anyway, wdyt @aschempp ?

Comment (aschempp) [reply modules/ModuleNewsletterUnsubscribeNotificationCenter.php]: we could require 3.5, but what advantage would that be? I'd rather keep it as is and then rewrite at some point for Contao 4...

File: config/config.php [hunk 2]
```diff
@@ -48,6 +48,12 @@
  */
 $GLOBALS['FE_MOD']['user']['lostPasswordNotificationCenter'] = 'ModulePasswordNotificationCenter';
 
+if (isset($GLOBALS['FE_MOD']['newsletter'])) {
```
Comment (aschempp) [inline config/config.php]: I think you should use `ModuleLoader::isActive('newsletter')` here

File: dca/tl_module.php [hunk 3]
```diff
@@ -15,6 +15,31 @@
 $GLOBALS['TL_DCA']['tl_module']['palettes']['registration'] = str_replace('reg_activate;', 'reg_activate,nc_notification,nc_activation_notification;', $GLOBALS['TL_DCA']['tl_module']['palettes']['registration']);
 $GLOBALS['TL_DCA']['tl_module']['palettes']['lostPasswordNotificationCenter'] = str_replace('reg_password', 'nc_notification', $GLOBALS['TL_DCA']['tl_module']['palettes']['lostPassword']);
 
+$GLOBALS['TL_DCA']['tl_module']['palettes']['newsletterSubscribeNotificationCenter'] = '{title_legend},name,headline,type'
```
Comment (aschempp) [inline dca/tl_module.php]: please don't use newlines in DCA palettes

File: config/config.php (hunk 2 above)
Comment (dmolineus) [reply config/config.php]: Changed in 9678fd6.

**Comment by dmolineus** [comment]
I adressed mentioned coding style issues and requested changes and did some fixes (missing translations, captcha issue). Pull request is sucessfully tested in Contao 3.5 and 4.4.

**Comment by aschempp** [comment]
Looks very good! Just one question before we merge this: do **you** need support for Contao 3.5? Otherwise we could only add the module if the user has Contao 4.x which means we need a lot less duplicate code …

**Comment by dmolineus** [comment]
> Thanks for the update @dmolineus. Do you think is is RTM now?

Yes it is, we're using it already in production.

**Pull request merged**